kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,111 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-08T00:00:00Z
+---
+
+All requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+
+`Context: deadline exceeded` is an error that occurs when requests are not completed on time.
+Network interruptions, short timeouts, server overload, and Query errors can cause this error.
+
+The following sections will discuss the nature of this error and how to troubleshoot it.
+
+#### Check your logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+If you're using Temporal Cloud, consider filing a service ticket when this error occurs.
+
+:::
+
+```
+{“level”:“error”,“ts”:“2022-03-21T19:32:42.312Z”,“msg”:“unavailable error”,“service”:“frontend”,“error”:“unable to get temporal-sys-add-search-attributes-workflow workflow state: context deadline exceeded”,“logging-call-at”:“adminHandler.go:1163”,“stacktrace”:“go.temporal.io/server/common/log.
+```
+
+Service logs can show which parts of the Cluster aren't working.
+
+Verify that the frontend is connected by opening the Web UI in your browser.
+
+Check that the following are up and running with `tctl cluster` commands:
+
+- [History Service](/concepts/what-is-a-history-service)
+- [Persistence](/clusters#persistence) database
+- [Advanced Visibility](/concepts/what-is-advanced-visibility) database
+
+Logs can also be used to find [Query](/concepts/what-is-a-query) errors.
+
+#### Check your Cluster metrics
+
+Look for high latencies, short timeouts, and other abnormal [Cluster metrics](/references/cluster-metrics).
+If the metrics come from a specific service (such as History Service), check that service's connectivity and restart if needed.
+
+#### Check Workflow logic
+
+Your Workflow may be missing some information needed to make the connection.
```

Comment:
workflow code doesn't make connections. not sure what types of logic result in deadline exceeded, maybe query handlers that have errors or are long running.